sparse_strips/vello_common/src/coarse.rs

```diff
@@ -71,46 +87,78 @@ struct Clip {
     pub thread_idx: u8,
 }
 
-/// A bounding box
+/// An axis-aligned bounding box represented by top-left and bottom-right corners.
 ///
-/// The first two values represent the x0 and y0 coordinates, respectively.
-/// The last two values represent the x1 and y1 coordinates, respectively.
-///  x0, y0 — the top-left corner of the bounding box,
-///  x1, y1 — the bottom-right corner of the bounding box.   
-#[derive(Debug, Clone)]
-struct Bbox {
+/// The coordinates are stored as `[x0, y0, x1, y1]` in wide tile coordinates,
+/// where `(x0, y0)` is the top-left corner and `(x1, y1)` is the bottom-right corner.
+#[derive(Debug, Clone, Copy)]
+pub struct Bbox {
+    /// The bounding box coordinates.
     pub bbox: [u16; 4],
 }
 
 impl Bbox {
-    pub(crate) fn new(bbox: [u16; 4]) -> Self {
+    /// Create a new bounding box.
+    pub fn new(bbox: [u16; 4]) -> Self {
         Self { bbox }
     }
 
     /// Get the x0 coordinate of the bounding box.
     #[inline]
-    pub(crate) fn x0(&self) -> u16 {
+    pub fn x0(&self) -> u16 {
         self.bbox[0]
     }
 
     /// Get the y0 coordinate of the bounding box.
     #[inline]
-    pub(crate) fn y0(&self) -> u16 {
+    pub fn y0(&self) -> u16 {
         self.bbox[1]
     }
 
     /// Get the x1 coordinate of the bounding box.
     #[inline]
-    pub(crate) fn x1(&self) -> u16 {
+    pub fn x1(&self) -> u16 {
         self.bbox[2]
     }
 
     /// Get the y1 coordinate of the bounding box.
     #[inline]
-    pub(crate) fn y1(&self) -> u16 {
+    pub fn y1(&self) -> u16 {
         self.bbox[3]
     }
 
+    /// Get the width of the bounding box (x1 - x0).
+    #[inline]
+    pub fn width_tiles(&self) -> u16 {
+        self.x1().saturating_sub(self.x0())
+    }
+
+    /// Get the width of the bounding box in pixels.
+    #[inline]
+    pub fn width(&self) -> u16 {
+        self.width_tiles() * WideTile::WIDTH
+    }
+
+    /// Get the height of the bounding box (y1 - y0).
+    #[inline]
+    pub fn height_tiles(&self) -> u16 {
+        self.y1().saturating_sub(self.y0())
+    }
+
+    /// Get the height of the bounding box in pixels.
+    #[inline]
+    pub fn height(&self) -> u16 {
+        self.height_tiles() * Tile::HEIGHT
+    }
+
+    /// Check if a point (x, y) is contained within this bounding box.
+    ///
+    /// Returns `true` if x0 <= x < x1 and y0 <= y < y1.
+    #[inline]
+    pub fn contains(&self, x: u16, y: u16) -> bool {
+        x >= self.x0() && x < self.x1() && y >= self.y0() && y < self.y1()
```

Comment:
It's probably worth making this branchless. The compiled form retains the branching. See the compilation [here](https://godbolt.org/z/zYMv99oWe)


```suggestion
        let [x0, y0, x1, y1] = self.bbox;
        (x >= x0) & (x < x1) & (y >= y0) & (y < y1)
```